src/coreclr/jit/hwintrinsiccodegenxarch.cpp

```diff
@@ -1252,22 +1252,31 @@ void CodeGen::genBaseIntrinsic(GenTreeHWIntrinsic* node)
         }
 
         case NI_Vector128_get_AllBitsSet:
-        case NI_Vector256_get_AllBitsSet:
-        {
             assert(op1 == nullptr);
-            if (varTypeIsFloating(baseType) && compiler->compOpportunisticallyDependsOn(InstructionSet_AVX))
+            if (compiler->compOpportunisticallyDependsOn(InstructionSet_AVX))
```

Comment:
This `if` shouldn't be needed. `emitIns_SIMD_R_R_R` should already be handling VEX vs non VEX